### PR TITLE
Fix explanation evaluation, plausibility measures and visualization, update example notebook

### PR DESCRIPTION
**Pull request by emanuele-moscato**
[WIP]

Fixes (fixing exceptions raised in the `examples/benchmark.ipynb` notebook):
- Use keyword arguments explicitly when creating an `ExplanationWithRationale` object by adding a rationale to an `Explanation` object.
- Pass the full `BaseEvaluator` instance when creating the `EvaluationMetricOutput` for the `Tokenf1_PlausibilityEvaluation` evaluation metric.
- Don't look for the `BEST_VALUE` attribute when visualizing plausibility evaluation metric, as it's currently defined only for faithfulness evaluation metrics (*could we define it for both metrics families?*)
- Minor other fixes.

Still to fix:
- Calls to `Benchmark.evaluate_samples` require that all the instances in the dataset have a valid rationale: what should we do if it's not valid (e.g. it's a list of empty lists instead of a list of integers)?

### CODE REVIEW

File: examples/benchmark.ipynb
Comment (g8a9) [inline examples/benchmark.ipynb:1]: I would probably avoid to push changes to the notebook (it seems that outputs are in there).

**Review by g8a9 — CHANGES_REQUESTED** [review]
Apart from edits to the notebook, lgtm!

**Comment by g8a9** [comment]
> Calls to Benchmark.evaluate_samples require that all the instances in the dataset have a valid rationale: what should we do if it's not valid (e.g., it's a list of empty lists instead of a list of integers)?

Regarding this, I don't think it's *required* that all instances have a rationale. I believe data from the "thermostat" datasets have it, but others don't. In the current version, the code checks whether the "rationale" attribute is present, and if so, it adds it to the explanation output *before* running the evaluation.

The whole process is a bit hard to follow, though. If you have ideas on how to improve it, I'd be happy to hear them!

File: examples/benchmark.ipynb
Comment (emanuele-moscato) [reply examples/benchmark.ipynb:1]: I think the previously committed version of the notebook shows the outputs, while I deleted them here. I'll revert to the old version, but I think there's one small fix to be made in a couple of cells.

**Review by g8a9 — APPROVED** [review]
lgtm
